src/main/java/mSearch/tool/FileSize.java

```diff
@@ -1,106 +1,67 @@
-        int retCode;
+    /**
+     * Return the size of a URL in bytes.
+     *
+     * @param url URL as String to query.
+     * @return size in bytes or -1.
+     */
+    private static long getFileSizeFromUrl(String url) {
         if (!url.toLowerCase().startsWith("http")) {
-            return ret;
+            return -1;
         }
-////        FilmeSuchen.listeSenderLaufen.inc(ssender, RunSender.Count.GET_SIZE_SUM);
-        try {
-            HttpURLConnection conn = (HttpURLConnection) new URL(url).openConnection();
-            conn.setRequestProperty("User-Agent", Config.getUserAgent());
-            conn.setReadTimeout(TIMEOUT);
-            conn.setConnectTimeout(TIMEOUT);
-            retCode = conn.getResponseCode();
-            if (retCode < 400) {
-                ret = conn.getContentLengthLong(); //gibts erst seit jdk 7
-            }
-            conn.disconnect();
 
-            // dann über eine Proxy
-            if (retCode == 403) {
-////                FilmeSuchen.listeSenderLaufen.inc(ssender, RunSender.Count.GET_SIZE_SUM403);
-//                if (!Config.proxyUrl.isEmpty() && Config.proxyPort > 0) {
-//                    // nur dann verwenden, wenn ein Proxy angegeben
-//                    try {
-//                        SocketAddress saddr = new InetSocketAddress(Config.proxyUrl, Config.proxyPort);
-//                        Proxy proxy = new Proxy(Proxy.Type.SOCKS, saddr);
-//                        conn = (HttpURLConnection) new URL(url).openConnection(proxy);
-//                        conn.setRequestProperty("User-Agent", Config.getUserAgent());
-//                        conn.setReadTimeout(TIMEOUT);
-//                        conn.setConnectTimeout(TIMEOUT);
-//                        ret = conn.getContentLengthLong(); //gibts erst seit jdk 7
-//                        conn.disconnect();
-//                        if (ret > 0) {
-//////                            FilmeSuchen.listeSenderLaufen.inc(ssender, RunSender.Count.GET_SIZE_PROXY);
-//                        }
-//                    } catch (Exception ex) {
-//                        ret = -1;
-//                        Log.errorLog(963215478, ex);
-//                    }
-//                }
-            }
-        } catch (Exception ex) {
-            ret = -1;
-            if (ex.getMessage().equals("Read timed out")) {
-                Log.errorLog(825141452, "Read timed out: " + ssender + " url: " + url);
-            } else {
-                Log.errorLog(643298301, ex, "url: " + url);
-            }
+        final Request request = new Request.Builder().url(url).head().build();
+        long respLength = -1;
+        try (Response response = MVHttpClient.getInstance().getReducedTimeOutClient().newCall(request).execute();
+             ResponseBody body = response.body()) {
+            if (response.isSuccessful())
+                respLength = body.contentLength();
+        } catch (IOException ex) {
+            ex.printStackTrace();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)